deployment/pytorch/torch-hub-yolov5.ipynb

```diff
@@ -0,0 +1,886 @@
+        "\n",
+        "from verta import Client\n",
+        "from verta.endpoint.autoscaling import Autoscaling\n",
+        "from verta.endpoint.autoscaling.metrics import CpuUtilizationTarget, MemoryUtilizationTarget, RequestsPerWorkerTarget\n",
+        "from verta.endpoint.resources import Resources\n",
+        "from verta.environment import Python\n",
+        "from verta.registry import VertaModelBase"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "LB20pSYuXcCD"
+      },
+      "source": [
+        "## 3. Verta Set Up"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "id": "hGwbnFGlXgJh"
+      },
+      "outputs": [],
+      "source": [
+        "os.environ['VERTA_HOST'] = ''\n",
+        "os.environ['VERTA_EMAIL'] = ''\n",
+        "os.environ['VERTA_DEV_KEY'] = ''"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "colab": {
+          "base_uri": "https://localhost:8080/"
+        },
+        "id": "wVcbn1_3Xhtm",
+        "outputId": "690f7584-b7cf-45fd-b2db-e0306573c0d5"
+      },
+      "outputs": [],
+      "source": [
+        "client = Client(os.environ['VERTA_HOST'])"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "5bBHXvKnX_Je"
+      },
+      "source": [
+        "## 4. Model Class"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "id": "o3YNlqVMEi6c"
+      },
+      "outputs": [],
+      "source": [
+        "class Predictor(VertaModelBase):\n",
+        "    def __init__(self, artifacts=None):\n",
+        "        self.model = torch.hub.load('ultralytics/yolov5', 'yolov5s', pretrained=True, trust_repo=True)\n",
+        "        \n",
+        "    def predict(self, urls):\n",
+        "        results = []\n",
+        "\n",
+        "        for url in urls:\n",
+        "            result = self.model(url)\n",
+        "            file_name, img_data = result.files[0], result.pandas().xyxy[0]\n",
+        "            results.append([file_name, img_data])\n",
+        "\n",
+        "        return results\n",
+        "\n",
+        "    def describe(self):\n",
+        "        return {\n",
+        "            'method': 'predict',\n",
+        "            'args': '[[url]]',\n",
+        "            'returns': '[[file_name, [xmin\tymin\txmax\tymax\tconfidence\tclass\tname]]]',\n",
+        "            'description': 'Identifies objects present in the images.',\n",
+        "            'input_description': 'List of image URLs.',\n",
+        "            'output_description': 'List with identified objects information.'\n",
+        "        }\n",
+        "        \n",
+        "    def example(self):\n",
+        "        return [['http://www.cs.ucf.edu/~aroshan/index_files/Dataset_PitOrlManh/images/000001_0.jpg']]"
```

Comment:
Here is an idea, but feel free to disregard if it seems too complex:

`ultralytics/yolov5` itself is able to take a list of URLs. So technically, I think we could write

```python
    def example(self):
        return [
            'http://www.cs.ucf.edu/~aroshan/index_files/Dataset_PitOrlManh/images/000001_0.jpg',
            'http://www.cs.ucf.edu/~aroshan/index_files/Dataset_PitOrlManh/images/000001_1.jpg',
        ]
```

and give that list directly to `self.model()`, without a `for` loop:

```python
    def predict(self, urls):
        result = self.model(urls)
        file_names, img_data = result.files, result.pandas().xyxy
        img_data = [boxes_df.to_dict("records") for boxes_df in img_data]

        return list(zip(file_names, img_data))
```

though we would need to make some changes to the rest of the notebook, where we do `results[0]`

Reply:
Not disregarded, done :+1: